### PR TITLE
feat: upgrade minimum php version

### PR DESCRIPTION
**Pull request by gasspper**
- the minimum version of php was raised to 8.2
- all variables were typed in the sdk

~The tests will be updated in another PR, it will be attached here as soon as it is uploaded~
The tests are corrected in the following PR: https://github.com/TransbankDevelopers/transbank-sdk-php/pull/286
image

### CODE REVIEW

File: src/Webpay/Options.php [hunk 1]
```diff
@@ -19,41 +25,48 @@ class Options
...
+    public ?string $commerceCode = null;
     /**
      * @var string Sets the environment that the SDK is going
      *             to point to (eg. TEST, LIVE, etc).
      */
-    public $integrationType = self::ENVIRONMENT_INTEGRATION;
-
-    public function __construct($apiKey, $commerceCode, $integrationType, $timeout = self::DEFAULT_TIMEOUT)
-    {
+    public string $integrationType = self::ENVIRONMENT_INTEGRATION;
```
Comment (mastudillot) [inline src/Webpay/Options.php:44]: remove value or set to null

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php [hunk 2]
```diff
@@ -6,39 +6,38 @@
...
     {
         return $this->installmentsAmount;
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getIdQueryInstallments()
+    public function getIdQueryInstallments(): string
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php:31]: add `?` in return

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php [hunk 3]
```diff
@@ -6,39 +6,38 @@
...
     {
         return $this->idQueryInstallments;
     }
 
     /**
-     * @return mixed
+     * @return array
      */
-    public function getDeferredPeriods()
+    public function getDeferredPeriods(): array
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php:39]: add `?` in return

File: src/PatpassComercio/Responses/InscriptionFinishResponse.php [hunk 4]
```diff
@@ -26,19 +27,18 @@ public function __construct($httpCode)
     }
 
     /**
-     * @return mixed
+     * @return ?string
      */
-    public function getStatus()
+    public function getStatus(): string
```
Comment (mastudillot) [inline src/PatpassComercio/Responses/InscriptionFinishResponse.php]: add `?` in return

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionStatusResponse.php [hunk 5]
```diff
@@ -6,75 +6,73 @@
...
 
     /**
-     * @return mixed|null
+     * @return string
      */
-    public function getCardNumber()
+    public function getCardNumber(): string
     {
         return $this->cardNumber;
     }
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/MallTransactionStatusResponse.php:78]: add `?` in return

File: src/Webpay/TransaccionCompleta/Responses/TransactionCreateResponse.php [hunk 6]
```diff
@@ -6,19 +6,18 @@
...
     {
         $this->token = Utils::returnValueIfExists($json, 'token');
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getToken()
+    public function getToken(): string
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/TransactionCreateResponse.php:19]: add `?` in return

File: src/Webpay/TransaccionCompleta/Responses/TransactionInstallmentsResponse.php [hunk 7]
```diff
@@ -6,39 +6,38 @@
...
     {
         return $this->idQueryInstallments;
     }
 
     /**
-     * @return mixed
+     * @return array
      */
-    public function getDeferredPeriods()
+    public function getDeferredPeriods(): array
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/TransactionInstallmentsResponse.php:39]: add `?` in return

File: src/Webpay/TransaccionCompleta/Responses/TransactionRefundResponse.php [hunk 8]
```diff
@@ -24,51 +24,50 @@ public function __construct($json)
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getType()
+    public function getType(): string
```
Comment (mastudillot) [inline src/Webpay/TransaccionCompleta/Responses/TransactionRefundResponse.php:29]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php [hunk 9]
```diff
@@ -65,75 +62,74 @@ public function isApproved()
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getBuyOrder()
+    public function getBuyOrder(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php:67]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php [hunk 10]
```diff
@@ -65,75 +62,74 @@ public function isApproved()
...
     {
         return $this->expirationDate;
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getAccountingDate()
+    public function getAccountingDate(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php:99]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php [hunk 11]
```diff
@@ -65,75 +62,74 @@ public function isApproved()
...
     {
         return $this->accountingDate;
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getTransactionDate()
+    public function getTransactionDate(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php:107]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php [hunk 12]
```diff
@@ -65,75 +62,74 @@ public function isApproved()
...
     {
         return $this->details;
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getVci()
+    public function getVci(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php:123]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/TransactionCaptureResponse.php [hunk 13]
```diff
@@ -10,56 +10,58 @@ class TransactionCaptureResponse
...
     {
         return $this->capturedAmount;
     }
 
     /**
-     * @return mixed
+     * @return int
      */
-    public function getResponseCode()
+    public function getResponseCode(): int
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/TransactionCaptureResponse.php:63]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php [hunk 14]
```diff
@@ -47,58 +47,60 @@ public function __construct($json)
...
     {
         return $this->balance;
     }
 
     /**
-     * @return mixed|null
+     * @return ?int
      */
-    public function getResponseCode()
+    public function getResponseCode(): int
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php:78]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php [hunk 15]
```diff
@@ -47,58 +47,60 @@ public function __construct($json)
...
-        return (int) $this->responseCode;
+        return $this->responseCode;
     }
 
     /**
-     * @return mixed|null
+     * @return string
      */
-    public function getType()
+    public function getType(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php:86]: add `?` in return

File: src/Webpay/WebpayPlus/Responses/TransactionStatusResponse.php [hunk 16]
```diff
@@ -23,38 +23,38 @@ public function __construct($json)
...
                 return false;
         }
     }
 
     /**
-     * @return mixed
+     * @return string
      */
-    public function getVci()
+    public function getVci(): string
```
Comment (mastudillot) [inline src/Webpay/WebpayPlus/Responses/TransactionStatusResponse.php:47]: add `?` in return

**Review by mastudillot — CHANGES_REQUESTED** [review]
Check my comments

File: src/Webpay/WebpayPlus/Responses/TransactionStatusResponse.php (hunk 16 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/TransactionStatusResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php (hunk 15 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php (hunk 14 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/TransactionRefundResponse.php]: done

File: src/Utils/RequestServiceTrait.php [hunk 17]
```diff
@@ -11,20 +12,20 @@
...
+     * @param string  $endpoint
+     * @param array  $payload
      *
-     * @throws \Transbank\Webpay\Exceptions\WebpayRequestException
+     * @throws WebpayRequestException
      *
      * @return mixed
      */
-    public function sendRequest($method, $endpoint, $payload = [])
+    public function sendRequest(string $method, string $endpoint, array $payload = [])
```
[the remark this answers is not in the document]
Comment (gasspper) [reply src/Utils/RequestServiceTrait.php]: done

File: src/Webpay/Options.php (hunk 1 above)
Comment (gasspper) [reply src/Webpay/Options.php]: done

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php (hunk 2 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php]: done

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php (hunk 3 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/MallTransactionInstallmentsResponse.php]: done

File: src/PatpassComercio/Responses/InscriptionFinishResponse.php (hunk 4 above)
Comment (gasspper) [reply src/PatpassComercio/Responses/InscriptionFinishResponse.php]: done

File: src/Webpay/TransaccionCompleta/Responses/MallTransactionStatusResponse.php (hunk 5 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/MallTransactionStatusResponse.php]: done

File: src/Webpay/TransaccionCompleta/Responses/TransactionInstallmentsResponse.php (hunk 7 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/TransactionInstallmentsResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php (hunk 9 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php]: done

File: src/Webpay/TransaccionCompleta/Responses/TransactionRefundResponse.php (hunk 8 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/TransactionRefundResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php (hunk 10 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php (hunk 11 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php (hunk 12 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/MallTransactionStatusResponse.php]: done

File: src/Webpay/WebpayPlus/Responses/TransactionCaptureResponse.php (hunk 13 above)
Comment (gasspper) [reply src/Webpay/WebpayPlus/Responses/TransactionCaptureResponse.php]: done

File: src/Webpay/TransaccionCompleta/Responses/TransactionCreateResponse.php (hunk 6 above)
Comment (gasspper) [reply src/Webpay/TransaccionCompleta/Responses/TransactionCreateResponse.php]: done

**Review by Matiasnickolas — APPROVED** [review]
LGTM 🤙

**Review by tribiec — APPROVED** [review]
GJ 🚀

**Review by tribiec — APPROVED** [review]

**Review by Matiasnickolas — APPROVED** [review]
LGTM 🌝

**Pull request merged**